tests/e2e/util.go

```diff
@@ -4915,3 +4915,97 @@ func waitForVolumeSnapshotContentToBeDeleted(client snapclient.Clientset, ctx co
 	})
 	return waitErr
 }
+
+/*
+This method will fetch the master node IP where controller is running
+*/
+func getK8sMasterNodeIPWhereControllerLeaderIsRunning(ctx context.Context,
+	client clientset.Interface, sshClientConfig *ssh.ClientConfig,
+	controller_name string) (string, string, error) {
+	ignoreLabels := make(map[string]string)
+	var k8sMasterNodeIP string
+	csiControllerPod := ""
+	csiPods, err := fpod.GetPodsInNamespace(client, csiSystemNamespace, ignoreLabels)
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	k8sMasterIPs := getK8sMasterIPs(ctx, client)
+	k8sMasterIP := k8sMasterIPs[0]
+	for _, csiPod := range csiPods {
+		if strings.Contains(csiPod.Name, vSphereCSIControllerPodNamePrefix) {
+			grepCmdForFindingCurrentLeader := "kubectl logs " + csiPod.Name + " -n " +
+				"vmware-system-csi " + controller_name + " | grep 'new leader detected, current leader:' " +
+				" | tail -1 | awk '{print $10}' | tr -d '\n'"
+			framework.Logf("Invoking command '%v' on host %v", grepCmdForFindingCurrentLeader,
+				k8sMasterIP)
+			result, err := sshExec(sshClientConfig, k8sMasterIP,
+				grepCmdForFindingCurrentLeader)
+			if err != nil || result.Code != 0 {
+				fssh.LogResult(result)
+				return "", "", fmt.Errorf("couldn't execute command: %s on host: %v , error: %s",
+					grepCmdForFindingCurrentLeader, k8sMasterIP, err)
+			}
+			csiControllerPod = result.Stdout
+			if csiControllerPod != "" {
+				break
+			}
+		}
+	}
+	framework.Logf("CSI pod %s where leader is running", csiControllerPod)
+	podData, err := client.CoreV1().Pods(csiSystemNamespace).Get(ctx, csiControllerPod, metav1.GetOptions{})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	masterNodeName := podData.Spec.NodeName
+	framework.Logf("Master node name %s where leader is running", masterNodeName)
+	k8sNodes, err := client.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	gomega.Expect(err).NotTo(gomega.HaveOccurred())
+	for _, node := range k8sNodes.Items {
+		if node.Name == masterNodeName {
+			addrs := node.Status.Addresses
+			for _, addr := range addrs {
+				if addr.Type == v1.NodeInternalIP && (net.ParseIP(addr.Address)).To4() != nil {
+					k8sMasterNodeIP = addr.Address
+					break
+				}
+			}
+		}
+	}
+	framework.Logf("Master node ip %s where leader is running", k8sMasterIP)
+	return csiControllerPod, k8sMasterNodeIP, nil
+}
+
+/*
+This method will kill container on the master node IP where controller is running
+*/
+func executeDockerPauseKillCmd(sshClientConfig *ssh.ClientConfig, k8sMasterNodeIP string,
```

Comment:
same here fix comment format,
update func name to match its actions
`controller_name` name should be `container` right?